### PR TITLE
RD2: Create an error record on Elevate API error response

### PR DESCRIPTION
**Pull request by andrewyu-salesforce**
As a user I want to be presented with clear error messages when commitment either cannot be created or if created, cannot be connected to the Recurring Donation.

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by andrewyu-salesforce** [comment]
Tracking: [W-8044403](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008cjUnIAI/view)

**Comment by andrewyu-salesforce** [comment]
@RadSFDeveloper @npsp-reedestockton The PR is ready for review

File: src/classes/RD2_EntryFormController_TEST.cls [hunk 1]
```diff
@@ -365,6 +366,38 @@ private with sharing class RD2_EntryFormController_TEST {
...
+
+        Test.startTest();
+        UTIL_Http_TEST.mockBadRequestCalloutResponse();
+
+        String jsonResponse = RD2_EntryFormController.createCommitment(rd.Id, PAYMENT_METHOD_TOKEN);
+        Test.stopTest();
+
+        List<Error__c> errors = errorGateway.getRecords();
+        System.assertEquals(1, errors.size(),
+            'one error record should be created when the elevate returns error response');
```
Comment (RadSFDeveloper) [inline src/classes/RD2_EntryFormController_TEST.cls:397]: `An error record should be created for an error response.`

File: src/classes/RD2_EntryFormController_TEST.cls [hunk 2]
```diff
@@ -365,6 +366,38 @@ private with sharing class RD2_EntryFormController_TEST {
...
+        System.assert(errors[0].Full_Message__c.contains(UTIL_Http_TEST.BAD_REQUEST_MESSAGE),
+            'The error record should contains the message from the error response');
```
Comment (RadSFDeveloper) [inline src/classes/RD2_EntryFormController_TEST.cls:399]: `The error record should contain error response message`

File: src/classes/RD2_EntryFormController_TEST.cls [hunk 3]
```diff
@@ -365,6 +366,38 @@ private with sharing class RD2_EntryFormController_TEST {
...
+            .build();
+        insert rd;
+
+        Test.startTest();
+        UTIL_Http_TEST.mockBadRequestCalloutResponse();
+
+        String jsonResponse = RD2_EntryFormController.createCommitment(rd.Id, PAYMENT_METHOD_TOKEN);
+        Test.stopTest();
+
+        List<Error__c> errors = errorGateway.getRecords();
```
Comment (RadSFDeveloper) [inline src/classes/RD2_EntryFormController_TEST.cls:395]: Let's also verify the `Error__c.Related_Record_Id__c` equals to RD Id and `Context_Type__c` is "Elevate".

**Review by RadSFDeveloper — COMMENTED** [review]
Thanks for making changes @andrewyu-salesforce . Below is the feedback to minimize changes and introduces only required modifications. Let me know if there are questions.

**Comment by andrewyu-salesforce** [comment]
@RadSFDeveloper @npsp-reedestockton The code is updated and is ready for review

**Review by RadSFDeveloper — APPROVED** [review]
Looks good, thank you @andrewyu-salesforce !

**Review by npsp-reedestockton — APPROVED** [review]
Looks great @andrewyu-salesforce!  Nice work.

**Comment by rponti-sforg** [comment]
QE testing is complete for this WI. Based on the WI Description, AC, PR comments, and Test Charter feedback, tests have passed.

Additional regression 'beta' testing has been done in a **spc-integration-npsp** org created off '**master**'. The same tests that were executed for this PR / WI passed in this context.

For further testing details, see Test Charter and WI:
https://salesforce.quip.com/B82mAbzBmU4E
https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008cjUnIAI/view

WI status has been set to 'Closed'.

**Pull request merged**